packages/api-catalog-service/src/app.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable no-underscore-dangle */
+/* eslint-disable no-await-in-loop */
+import { ApolloServer, FastifyContext } from 'apollo-server-fastify';
+import { ApolloServerPluginDrainHttpServer, ContextFunction } from 'apollo-server-core';
+import { ApolloServerPlugin } from 'apollo-server-plugin-base';
+import fastify, { FastifyInstance } from 'fastify';
+import { IExecutableSchemaDefinition, mergeSchemas } from '@graphql-tools/schema';
+import { GraphQLSchema } from 'graphql';
+import mongoose from 'mongoose';
+import { Logger } from 'pino';
+
+import { Config } from 'config';
+import { setupUserDataLoader, setupSubscriptionStatusLoader } from 'dataloaders';
+
+import { NamespaceDB } from 'datasources/namespaceDB';
+import { SubscriptionDB } from 'datasources/subscriptionDB';
+import { SpecStoreDB } from 'datasources/specstoreDB';
+import { Namespace } from 'db/namespace';
+import { Subscription } from 'db/subscription';
+import { SpecStore } from 'db/specStore';
+
+import NamespaceResolver from './graph/namespace/resolver';
+import NamespaceSchema from './graph/namespace/typedef.graphql';
+import { SubscriptionResolvers } from './graph/subscription/resolver';
+import SubscriptionSchema from './graph/subscription/typedef.graphql';
+import SharedSchema from './graph/typedef.graphql';
+
+export const setupMongoose = async (mongoURI: string, logger: Logger) => {
+  /* Setup database connection */
+  try {
+    await mongoose.connect(mongoURI);
+    logger.info('Connected to database');
+  } catch (error) {
+    logger.error(error, 'Failed to connect to database');
+    process.exit(0);
```

Comment:
Use exit status code `1` on failure.
```suggestion
    process.exit(1);
```

Reply:
Changed